kernel/apic/src/lib.rs

```diff
@@ -39,22 +41,22 @@ pub enum InterruptChip {
 const_assert!(AtomicCell::<InterruptChip>::is_lock_free());
 
 /// The set of system-wide `LocalApic`s, one per CPU core.
-static LOCAL_APICS: AtomicMap<u8, RwLockIrqSafe<LocalApic>> = AtomicMap::new();
+static LOCAL_APICS: AtomicMap<CoreId, RwLockIrqSafe<LocalApic>> = AtomicMap::new();
 
 /// The number of CPUs currently initialized in the system.
 /// This must match the number of Local APICs initialized in the system.
-static CPU_COUNT: AtomicU8 = AtomicU8::new(0);
+static CPU_COUNT: AtomicUsize = AtomicUsize::new(0);
```

Comment:
u32 is the max size of a CPU ID
```suggestion
static CPU_COUNT: AtomicU32 = Atomic32::new(0);
```